framework/include/materials/MaterialPropertyInterface.h

```diff
@@ -546,7 +565,39 @@ MaterialPropertyInterface::hasMaterialPropertyByName(const std::string & name)
 
 template <typename T, bool is_ad>
 const GenericMaterialProperty<T, is_ad> &
-MaterialPropertyInterface::getGenericZeroMaterialProperty(const std::string & /*prop_name*/)
+MaterialPropertyInterface::getGenericZeroMaterialProperty(const std::string & name)
+{
+  std::string prop_name = deducePropertyName(name);
```

Comment:
The bits in `deducePropertyName(name)` is replicated in MaterialBase.h. I did it for declareGeneric, and you're doing it in this PR. Is there a better place it can sit so code doesn't have to be duplicated?

Reply:
`deducePropertyName` could probably be in `MaterialPropertyInterface.h` only. This situation precipitated a while back when MaterialBase was factored out and Material was made to inherit from MaterialPropertyInterface, which it didn't before as far as I remember.

Reply:
Actually, there is only one implementation of `deducePropertyName`, so I don't quite understand what duplication you want to avoid, and how.

Reply:
It's using deducePropertyName instead of the extra bits in the other comment